src/solvers/refinement/string_refinement_util.h

```diff
@@ -140,4 +141,203 @@ class equation_symbol_mappingt
   std::unordered_map<std::size_t, std::vector<exprt>> strings_in_equation;
 };
 
+/// Base class for string functions that are built in the solver
+class string_builtin_functiont
+{
+public:
+  string_builtin_functiont() = default;
```

Comment:
If this is not intended to be instantiated, you could mark this constructor as `protected`.